src/collective/fingerpointing/browser/logview.py

```diff
@@ -32,10 +35,35 @@ def tail(self, f, n=100):
         return lines[-n:]
 
     @property
+    def _get_audit_log_files(self):
+        """Return an ordered list of log files to be processed. We need
+        to process the current audit log and all of its backups.
+        """
+        # get all audit log backups in the log directory
+        logdir = os.path.dirname(LOGFILE)
+        logfiles = os.listdir(logdir)
+        # filter the list and sort it in reverse order
+        # as we will need to process newer backup files first
+        logfiles = sorted([
+            f for f in logfiles
+            if fnmatch(f, 'audit.log.*') and '.lock' not in f
+        ], reverse=True)
+        # we need to use the fully qualified filename
+        logfiles = [os.path.join(logdir, f) for f in logfiles]
+        # include active audit log as first element
+        logfiles.insert(0, LOGFILE)
+        return logfiles
+
+    @property
     def get_audit_log(self):
         """Return audit log with newer entries first."""
-        with open(logfile, 'r') as fp:
-            lines = self.tail(fp)
+        lines = []
+        # process all logs until we have enough lines
+        for logfile in self._get_audit_log_files:
+            with open(logfile, 'r') as fp:
+                lines.extend(self.tail(fp))
+            if len(lines) >= 100:
+                break
 
-        lines.reverse()
+        lines = sorted(lines[:100], reverse=True)
```

Comment:
I would use 2 tricks here:

1) [Open file and reverse lines in one shot](http://stackoverflow.com/a/2301792/2116850)
2) Transform this method into a [python generator](https://wiki.python.org/moin/Generators)

This way you will not keel all log lines into memory